site/docs/contributing.md

```diff
@@ -0,0 +1,316 @@
+
+## Development Workflow
+
+1. Create a new branch for your feature or bug fix:
+
+   ```sh
+   git checkout -b feature/your-feature-name
+   ```
+
+2. We try to follow the [Conventional Commits](https://www.conventionalcommits.org/) specification. This is not required for feature branches. We merge all PRs into `main` with a squash merge and a conventional commit message.
+
+3. Push your branch to your fork:
+
+   ```sh
+   git push origin your-branch-name
+   ```
+
+4. Open a pull request against the `main` branch of the promptfoo repository.
+
+When opening a pull request:
+
+- Keep changes small and focused. Avoid mixing refactors with new features.
+- Ensure test coverage for new code or bug fixes.
+- Provide clear instructions on how to reproduce the problem or test the new feature.
+- Be responsive to feedback and be prepared to make changes if requested.
+- Ensure your tests are passing and your code is properly linted.
+
+Don't hesitate to ask for help. We're here to support you. If you're worried about whether your PR will be accepted, please talk to us first.
+
+## Tests
+
+### Running Tests
+
+We use Jest for testing. To run the test suite:
+
+```sh
+npm test
+```
+
+To run tests in watch mode:
+
+```sh
+npm run test:watch
+```
+
+You can also run specific tests with:
+
+```sh
+npx jest [pattern]
+```
+
+### Writing Tests
+
+When writing tests, please:
+
+- Run them with the `--randomize` flag to ensure your mocks setup and teardown are not affecting other tests.
+- Check the coverage report to ensure your changes are covered.
+- Avoid adding additional logs to the console.
+
+## Linting and Formatting
+
+We use ESLint and Prettier for code linting and formatting. Before submitting a pull request, please run:
+
+```sh
+npm run format
+npm run lint
+```
+
+It's a good idea to run the lint command as `npm run lint -- --fix` to automatically fix some linting errors.
+
+## Building the Project
+
+To build the project:
+
+```sh
+npm run build
+```
+
+For continuous building during development:
+
+```sh
+npm run build:watch
+```
+
+## Contributing to the CLI
+
+### Running the CLI During Development
+
+We recommend using `npm link` to link your local `promptfoo` package to the global `promptfoo` package:
+
+```sh
+npm link
+promptfoo --help
+```
+
+Alternatively, you can run the CLI directly:
+
+```sh
+npm run local -- eval --config examples/cloudflare-ai/chat_config.yaml
+```
+
+When working on a new feature, we recommend setting up a local `promptfooconfig.yaml` that tests your feature. Think of this as an end-to-end test for your feature.
+
+Here's a simple example:
+
+```yaml
+providers:
+  - name: openai:chat:gpt-4o
+prompts:
+  - Translate "{{input}}" to {{language}}
+tests:
+  - vars:
+      input: 'Hello, world!'
+      language: 'English'
+    assert:
+      - type: new-assertion-type
+```
+
+## Adding a New Provider
+
+1. Create an implementation in `src/providers/SOME_PROVIDER_FILE`.
```

Comment:
good to link to /docs/providers/custom-api/ for details on how to implement a typescript provider